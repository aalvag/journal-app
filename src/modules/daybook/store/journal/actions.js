// export const myAction = async ({ commit }, payload) => {}

import journalApi from "@/api/journalApi";

export const loadEntries = async ({ commit }) => {
  try {
    const { data } = await journalApi.get("/entries.json");

    if (!data) {
      commit("setEntries", []);
      return;
    }
    const entries = [];
    for (let id of Object.keys(data)) {
      entries.push({
        id,
        ...data[id],
      });
    }

    commit("setEntries", entries);
  } catch (error) {
    console.log(error);
  }
};

export const updateEntry = async ({ commit }, entry) => {
  try {
    const { id, ...dataToSAve } = entry;
    await journalApi.put(`/entries/${id}.json`, dataToSAve);

    commit("updateEntry", { ...entry });
  } catch (error) {
    console.log(error);
  }
};

export const createEntry = async ({ commit }, newEntry) => {
  try {
    const { data } = await journalApi.post(`/entries.json`, newEntry);
    newEntry.id = data.name;
    commit("addEntry", newEntry);
    return newEntry.id;
  } catch (error) {
    console.log(error);
  }
};

export const deleteEntry = async ({ commit }, id) => {
  try {
    await journalApi.delete(`/entries/${id}.json`);
    commit("deleteEntry", id);
  } catch (error) {
    console.log(error);
  }
};
