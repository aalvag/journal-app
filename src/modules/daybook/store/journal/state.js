export default () => ({
  isLoading: true,
  entries: [
    {
      id: 1,
      date: new Date().toDateString(),
      text: "Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum. Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.Hola mundo //#endregion Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.",
      picture: null,
    },
    {
      id: 2,
      date: new Date().toDateString(),
      text: "que tal",
      picture: null,
    },
    {
      id: 3,
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.",
      picture: null,
    },
  ],
});
