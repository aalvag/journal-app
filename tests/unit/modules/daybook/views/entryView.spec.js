import { createStore } from "vuex";

import journal from "@/modules/daybook/store/journal";
import { journalState } from "../../../mock-data/test-journal-state";
import { shallowMount } from "@vue/test-utils";
import EntryViewVue from "@/modules/daybook/views/EntryView.vue";
import Swal from "sweetalert2";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

jest.mock("sweetalert2", () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn(),
}));

describe("Daybook Entry View", () => {
  const store = createVuexStore(journalState);
  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryViewVue, {
      props: {
        id: "-MjGQ7Q0Z1c3i9hjRyZO",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  it("must redirect the user because the id does not exist.", () => {
    shallowMount(EntryViewVue, {
      props: {
        id: "this-id-does-not-exist",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });

    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });

  it("must display the entry correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("must delete the entry and redirect to the main page", async () => {
    Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }));
    const button = await wrapper.find(".btn-danger");
    await button.trigger("click");

    expect(Swal.fire).toHaveBeenCalledWith({
      title: "Are you sure?",
      text: "The entry will be deleted permanently",
      showCancelButton: true,
      confirmButtonText: "Yes, delete it!",
      cancelButtonText: "No, keep it",
      icon: "warning",
    });

    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });
});
