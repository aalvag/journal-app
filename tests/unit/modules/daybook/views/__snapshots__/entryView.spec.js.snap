// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Daybook Entry View must display the entry correctly 1`] = `
<!--v-if-->
<follow-action-button-stub icon="fas fa-save"></follow-action-button-stub>
<!--v-if-->
<!--v-if-->
`;
